src/pip/_internal/utils/outdated.py

```diff
@@ -119,12 +120,11 @@ def pip_version_check(session, options):
                 process_dependency_links=options.process_dependency_links,
                 session=session,
             )
-            all_candidates = finder.find_all_candidates("pip")
-            if not all_candidates:
+            try:
+                candidate = finder.find_best_candidate("pip", SpecifierSet())
```

Comment:
It seems like the caller shouldn't have to know about `SpecifierSet` objects to call `find_best_candidate()`. What about making that argument optional?

Reply:
I am more fond of being explicit :) Convenience is not as important here since the API is internal (anyone wanting to understand the this line need to know about specifiers anyway), and always requiring a `SpecifierSet` reduces the chance of forgetting to pass one when you should.

Would `finder.find_best_candidate("pip", specifier=None)` be an acceptable middle ground?

Reply:
I concur omitting the specifier should be okay, since `finder.find_best_candidate("pip")` is clear enough in terms of communicating what's happening.

I'm okay with all three options on the table, only slightly preferring the form I've stated here.